src/funcTest/groovy/info/solidsoft/gradle/pitest/functional/Junit5FunctionalSpec.groovy

```diff
@@ -58,4 +58,21 @@ class Junit5FunctionalSpec extends AbstractPitestFunctionalSpec {
             result.standardOutput.contains('Generated 1 mutations Killed 1 (100%)')
     }
 
+    @Issue("https://github.com/szpak/gradle-pitest-plugin/issues/249")
+    void "should store and reuse Gradle configuration cache"() {
+        given:
+            copyResources("testProjects/junit5simple", "")
+        when:
+            ExecutionResult result = runTasks('pitest', '--configuration-cache')
+            ExecutionResult result2 = runTasks('pitest', '--configuration-cache')
+        then:
+            !result.failure
+            result.wasExecuted('pitest')
+            !result2.failure
+            result2.wasExecuted('pitest')
+        and:
+            result.standardOutput.contains('Configuration cache entry stored')
+            result2.standardOutput.contains('Reusing configuration cache.')
```

Comment:
I'm not sure why the previous version of production code throws:
> ClassCastException: class java.lang.Object cannot be cast to class info.solidsoft.gradle.pitest.PitestPlugin  java.lang.Object is in module java.base of loader 'bootstrap'; info.solidsoft.gradle.pitest.PitestPlugin is in  nnamed module of loader org.gradle.internal.classloader.VisitableURLClassLoader @1d6d2b74)

but after refactoring the test seems to work :-/.

Reply:
Well, I'm no expert on the inner workings of the configuration cache, but I reason that in the prior version, `combinedTaskClasspath.filter` is fed with a lambda that refers to the outer `extension` field, which inherently forces the lambda to have a reference to the plugin object, and which subsequently fails deserializing from the cache. After the refactoring, the `List<String>` value is available standalone. The fact that it also is a slight performance improvement is just icing on the cake 🍰